### PR TITLE
Add support for setting routingConfig:subdomain

### PR DESCRIPTION
**Pull request by detiber**
Addresses https://github.com/openshift/openshift-ansible/issues/282

### CODE REVIEW

**Comment by detiber** [comment]
@sdodson @twiest @wshearn Could use a review on this.  I've tested it in the context of the byo playbooks (and the keynote demo productization script), but haven't extended it to be used by the other provider playbooks.

File: roles/openshift_master/templates/master.yaml.v1.j2 [hunk 1]
```diff
@@ -101,7 +101,7 @@ projectConfig:
     mcsLabelsPerProject: 5
     uidAllocatorRange: 1000000000-1999999999/10000
 routingConfig:
-  subdomain: router.default.local
+  subdomain:  "{{ openshift.master.default_subdomain | default("") }}"
```
Comment (wshearn) [inline roles/openshift_master/templates/master.yaml.v1.j2]: If it is not set it should default to the old value of router.default.local.

Comment (detiber) [inline roles/openshift_master/templates/master.yaml.v1.j2]: @wshearn I thought about that, but router.default.local doesn't match up with the upstream default of router.default.svc.cluster.local

**Comment by sdodson** [comment]
LGTM, I'm fine with changing the default as the default really wasn't usable to begin with.

**Comment by twiest** [comment]
:+1:

**Pull request merged**